client/packages/programs/src/JsonForms/components/IdGenerator.tsx

```diff
@@ -213,17 +217,34 @@ const generateId = async (input: GenerateIdInput): Promise<string> => {
 
 const UIComponent = (props: ControlProps) => {
   const { label, path, data, visible, handleChange, uischema, config } = props;
-  const t = useTranslation('common');
+  const t = useTranslation(['patients', 'common']);
   const { mutateAsync: mutateGenerateId } = useMutation(
     async (input: GenerateIdInput): Promise<string> => generateId(input)
   );
   const { mutateAsync: allocateNumber } = useDocument.utils.allocateNumber();
 
+  const { data: savedData } = useDocument.get.document(config.documentName);
+
   const { errors, options } = useZodOptionsValidation(
     GeneratorOptions,
     uischema.options
   );
 
+  /*
+  Regeneration behaviour:
+  - By default, after ID is first saved, a confirmation will be displayed
+    whenever the user subsequently clicks the "Generate" button
+  - This can be suppressed by setting the "confirmRegenerate" option to `false`
+    (default `true`)
+  - To prevent the ID from *ever* being regenerated after first save, set the
+    "preventRegenAfterSave" option to `true` (default `false`)
+  */
+  const canGenerate =
+    !options?.preventRegenAfterSave || !savedData?.data?.code2;
```

Comment:
If there is no data saved yet, then "canGenerate" should be true regardless of the setting (that's the `!savedData?.data?.code2` part). But if there _is_ data saved, then we consider the "preventRegenerate" setting. Hence "or".

I've switched them around to make it somewhat clearer:
```
const canGenerate = !savedData?.data?.code2 || !options?.preventRegenerate;
```

This might be a case of too many double-negatives which is always confusing.

Reply:
yes  sorry you are you are right